doc/ysh-faq.md

```diff
@@ -72,22 +72,24 @@ The rules are subtle, so we avoid inventing new ones.
 
 ## How do I write the equivalent of `echo -e` or `echo -n`?
 
-YSH removed these flags to `echo` because it has good alternatives
+YSH removed these flags to `echo` to solve their interference problem with
+variable substitutions (see next) and double-quoted backslash.
 
-To escape variables, use the statically-parsed string language, rather than
+To echo escape sequences, use a statically-parsed string literal, rather than
 `echo -e`:
 
-    echo $'tab \t newline \n'   # YES
-    echo j"tab \t newline \n"   # TODO: J8 notation
+    echo $'tab: \t newline: \n' # YES
+    echo u'tab: \t A: \u{41}'   # J8 unicode notation
 
-    echo -e tab \t newline \n'  # NO
+    echo -e "tab: \t newline: \n"    # NO (backslash)  => Invalid char escape
```

Comment:
I think these examples should use single quotes, not double quotes, because there is an extra meaning of `\\` in double quotes

Reply:
Hm, but it's the NO side showing problems, and there actually are two things involved in why echo does not work in the same way it does in shell. So, I find it only helpful to show both issues on separate example lines.

Also the string type supporting substitutions is probably more important for echo in general.

Reply:
In another way also "The example should use double quotes, because they support substitutions".

So actually, what's the ysh way to insert substitutions? `echo $'tab:\t'$name$'continued'`?

Because these don't work:
```
echo u'tab:\t'"$name"u'continued'
echo u'tab:\t'${name}u'continued'
```

Reply:
I think there may still be no proper equivalent to `echo -e "\t $var"`, or do you know one?

```
ysh ysh-0.20.0$ echo u'\t'$indented
  echo u'\t'$indented
            ^~~~~~~~~
[ interactive ]:1: Unexpected token after YSH single-quoted string
```

As of this, I could only use echo's default space separation of arguments as current workaround in the FAQ:

```
echo $'tab: \t' $indented  # YES
echo u'tab: \t' $indented  # J8 with \u{unicode} notation
echo b'tab: \t' $indented  # J8 with \u{unicode} and \y{byte} notation

echo -e "tab: \t $indented"   # NO (backslash)  => Error: Invalid char escape
echo -e "tab: \\t $indented"  # NO (echo_flags) => Prints: "-e tab..."
```

Reply:
With something before the u'-string, it' s not an error, but instead it's not recognized as such:
```
ysh ysh-0.20.0$ echo ${first}u'\t'$second
EXTENDEDu\tTEXT
ysh ysh-0.20.0$ 
```

Reply:
Added this one:
`echo $[ u'tab: \t' ++ indented ] # YES (generic YSH, not "IFS separated)`
Is that it?

Reply:
Yes, using `++` is a good way to combine different quote types!

That's useful in this section